delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/context/JsonLdContext.scala

```diff
@@ -0,0 +1,158 @@
+    )
+
+  /**
+    * @return the all the values with key @context
+    */
+  def contextValues(json: Json): Set[Json] =
+    json.extractValuesFrom(keywords.context)
+
+  /**
+    * @return the inner Json with the top @context key or and empty Json
+    */
+  def topContextValueOr(json: Json, default: => Json = Json.obj()): Json =
+    topContextValue(json).getOrElse(default)
+
+  /**
+    * Merges the values of the key @context in both passed ''json'' and ''that'' Json documents.
+    *
+   * @param json the primary context. E.g.: {"@context": {...}}
+    * @param that the context to append to this json. E.g.: {"@context": {...}}
+    * @return a new Json with the original json and the merged context of both passed jsons.
+    *         If a key inside the @context is repeated in both jsons, the one in ''that'' will override the one in ''json''
+    */
+  def addContext(json: Json, that: Json): Json =
+    json deepMerge Json.obj(keywords.context -> merge(topContextValueOr(json), topContextValueOr(that)))
+
+  /**
+    * Adds a context IRI to an existing JSON object.
+    */
+  def addContext(json: Json, contextIri: IRI): Json = {
+    val jUriString = Json.fromString(contextIri.toString)
+
+    json.asObject match {
+      case Some(obj) =>
+        val updated = obj(keywords.context) match {
+          case None           => obj.add(keywords.context, jUriString)
+          case Some(ctxValue) =>
+            (ctxValue.asObject, ctxValue.asArray, ctxValue.asString) match {
+              case (Some(co), _, _) if co.isEmpty                         => obj.add(keywords.context, jUriString)
+              case (_, Some(ca), _) if ca.isEmpty                         => obj.add(keywords.context, jUriString)
+              case (_, _, Some(cs)) if cs.isEmpty                         => obj.add(keywords.context, jUriString)
+              case (Some(co), _, _) if !co.values.exists(_ == jUriString) =>
+                obj.add(keywords.context, Json.arr(ctxValue, jUriString))
+              case (_, Some(ca), _) if !ca.contains(jUriString)           =>
+                obj.add(keywords.context, Json.fromValues(ca :+ jUriString))
+              case (_, _, Some(cs)) if cs != contextIri.toString          =>
+                obj.add(keywords.context, Json.arr(ctxValue, jUriString))
+              case _                                                      => obj
+            }
+        }
+        Json.fromJsonObject(updated)
+      case None      => json
+    }
+  }
+
+  private def removeEmpty(arr: Seq[Json]): Seq[Json] =
+    arr.filter(j => j != Json.obj() && j != Json.fromString("") && j != Json.arr())
+
+  private def arrOrObj(arr: Seq[Json]): Json =
+    arr.take(2).toList match {
+      case Nil         => Json.obj()
+      case head :: Nil => head
+      case _ :: _      => Json.arr(arr: _*)
+    }
+
+  private def merge(json: Json, that: Json): Json =
+    (json.asArray, that.asArray, json.asString, that.asString) match {
```

Comment:
Why the asString part ?

Reply:
To check the case of context being a string, like 
```json
{
   "@context": "http://schema.org/"
}
```